ci(r): check-c for ADBC SQLite on macOS fails since R 4.5.1

### What happened?

Recently, `check-c` for ADBC SQLite on macOS fails.

https://github.com/apache/arrow-adbc/actions/runs/15677308786/job/44160470753?pr=2975

It appears that it has failed since the R 4.5.1 release (2025-06-13).

`R CMD check` of https://github.com/r-lib/actions/tree/v2/check-r-package fails as follows.

```
 * checking whether package ‘adbcsqlite’ can be installed ... WARNING
Warning: Found the following significant warnings:
  tools/test_extension.c:21:29: warning: 'sqlite3_auto_extension' is deprecated: first deprecated in macOS 10.10 - Process-global auto extensions are not supported on Apple platforms [-Wdeprecated-declarations]
See ‘/Users/runner/work/arrow-adbc/arrow-adbc/r/adbcsqlite/check/adbcsqlite.Rcheck/00install.out’ for details.
```


### Stack Trace

None.

### How can we reproduce the bug?

Run `R (standard)` CI for ADBC SQLite on macOS.

### Environment/Setup

I have confirmed this warning does not occur with R 4.5.0.
The reason is unclear, but it appears to be caused by R 4.5.1.

```diff
diff --git a/.github/workflows/r-check.yml b/.github/workflows/r-check.yml
index 2746bc41f..b9b8d18df 100644
--- a/.github/workflows/r-check.yml
+++ b/.github/workflows/r-check.yml
@@ -53,7 +53,7 @@ jobs:
           go-version: "${{ env.GO_VERSION }}"
       - uses: r-lib/actions/setup-r@bd49c52ffe281809afa6f0fecbf37483c5dd0b93  # v2
         with:
-          r-version: release
+          r-version: '4.5.0'
           use-public-rspm: true
 
       - name: Set PKG_CONFIG_PATH on MacOS
```

--- a/r/adbcsqlite/cleanup
+++ b/r/adbcsqlite/cleanup
@@ -17,3 +17,4 @@
 
 find src -name "*.o" -delete || true
 find src -name "*.so" -delete || true
+rm sqlite_test.log || true

--- a/r/adbcsqlite/configure
+++ b/r/adbcsqlite/configure
@@ -59,7 +59,6 @@ else
 
   if [ $? -ne 0 ]; then
     echo "Compile of sqlite3_load_extension() test failed"
-    cat sqlite_test.log
     PKG_CPPFLAGS="$PKG_CPPFLAGS -DADBC_SQLITE_WITH_NO_LOAD_EXTENSION"
   else
     echo "Success!"
@@ -71,7 +70,7 @@ if [ ! -z "$R_ADBC_SQLITE_ON_WINDOWS" ]; then
   PKG_CPPFLAGS="$PKG_CPPFLAGS -D__USE_MINGW_ANSI_STDIO"
 fi
 
-rm -f tools/test.o tools/test_extension.o sqlite_test sqlite_test.log || true
+rm -f tools/test.o tools/test_extension.o sqlite_test || true
 
 sed \
   -e "s|@cppflags@|$PKG_CPPFLAGS|" \

